src/Dapr.AzureFunctions.Extension/Services/DaprServiceListener.cs

```diff
@@ -133,7 +133,9 @@ public bool Equals(DaprTopicSubscription topic1, DaprTopicSubscription topic2)
                 {
                     return false;
                 }
-                else if (topic1.Topic.Equals(topic2.Topic, StringComparison.OrdinalIgnoreCase)
+                else if (
+                    topic1.Name.Equals(topic2.Name, StringComparison.OrdinalIgnoreCase)
```

Comment:
Discussed offline with @yaron2 - topic and pub/sub name are actually case-sensitive.